doc/source/data/examples/batch_training.ipynb

```diff
@@ -0,0 +1,1228 @@
+       "98 2019-05-01 00:47:54 2019-05-01 00:53:01                 162   \n",
+       "99 2019-05-01 00:08:37 2019-05-01 00:13:51                 239   \n",
+       "\n",
+       "    dropoff_location_id  passenger_count  trip_distance  fare_amount  \n",
+       "0                   145                1           1.50          3.0  \n",
+       "1                   145                1           1.50          2.5  \n",
+       "2                   161                1           0.70          5.0  \n",
+       "3                   141                1           2.00          9.5  \n",
+       "4                    56                1           2.50         10.0  \n",
+       "..                  ...              ...            ...          ...  \n",
+       "95                  106                1          10.60          2.5  \n",
+       "96                  241                1          15.50         42.5  \n",
+       "97                   87                1           1.16          6.0  \n",
+       "98                  162                2           0.88          5.5  \n",
+       "99                  142                1           0.90          6.0  \n",
+       "\n",
+       "[100 rows x 7 columns]"
+      ]
+     },
+     "execution_count": 12,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "df = pushdown_ds.limit(100).to_pandas()\n",
+    "df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Custom data transform functions\n",
+    "\n",
+    "Ray Datasets allows you to specify custom data transform functions. These [user defined functions (UDFs)](transforming_datasets) can be called using `Dataset.map_batches(my_UDF)`. The transformation will be conducted in parallel for each data batch.\n",
+    "\n",
+    "```{tip}\n",
+    "You may need to call `Dataset.repartition(n)` first to split the Dataset into more blocks internally. By default, each block corresponds to one file. The upper bound of parallelism is the number of blocks.\n",
+    "```\n",
+    "\n",
+    "You can specify the data format you are using in the `batch_format` parameter. The dataset will be divided into batches and those batches converted into the specified format. Available data formats you can specify in the `batch_format` paramater include `\"pandas\", \"pyarrow\", \"numpy\"`. Tabular data will be passed into your UDF by default as a pandas DataFrame. Tensor data will be passed into your UDF as a numpy array.\n",
+    "\n",
+    "Here, we will use `batch_format=\"pandas\"` explicitly for clarity."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# A pandas DataFrame UDF for transforming the Dataset in parallel.\n",
+    "def transform_batch(df: pd.DataFrame) -> pd.DataFrame:\n",
+    "    df[\"trip_duration\"] = (df[\"dropoff_at\"] - df[\"pickup_at\"]).dt.seconds\n",
```

Comment:
Nit: Use the `.loc[]` API to mutate the DataFrame rather than create a new copy.

```suggestion
    "    df.loc[\"trip_duration\"] = (df[\"dropoff_at\"] - df[\"pickup_at\"]).dt.seconds\n",
```